sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/SonarAnalysisContext.cs

```diff
@@ -29,19 +30,32 @@ namespace SonarAnalyzer.Helpers
     public class SonarAnalysisContext

```

Comment:
This class is now stateful and specific to a single analyser. It's probably worth documenting this, which would make it clearer that the list of _supportedDiagnostics_ isn't an arbitrary list, it's just the rules for a specific analyser.

Reply:
It was already stateful (was taking the roslyn context). But I will add a comment.